libs/core/algorithms/include/hpx/parallel/algorithms/count.hpp

```diff
@@ -444,7 +444,8 @@ namespace hpx {
     {
     private:
         // clang-format off
-        template <typename ExPolicy, typename FwdIter, typename T,
+        template <typename ExPolicy, typename FwdIter,
+            typename T = typename std::iterator_traits<FwdIter>::value_type,
```

Comment:
Note: `std::iterator_traits` is not SFINAE friendly, i.e. compilation will fail for non-iterator types. We might have to create a SFINAE-friendly equivalent to avoid compilation errors.

Reply:
How would we do that? Create a wraparound facility that is SFINAE friendly that returns iterator_traits if it's an iterator else SFINAE fails?

Reply:
Something like:
```
template <typename Iter, typename Enable = std::enable_if_t<hpx::traits::is_iterator_v<Iter>>>
struct iterator_traits : std::iterator_traits<Iter> {};
```
(see: https://godbolt.org/z/Mc3cPW51e)